src/components/ChartComponent/ChartComponent.jsx

```diff
@@ -0,0 +1,115 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import CovidService from '../../api/Covid-service';
+import ChartComponentView from './ChartComponentView';
+
+export default class ChartComponent extends Component {
+  covidService = new CovidService();
+
+  constructor() {
+    super();
+    this.state = {
+      chartData: null,
+      isLoading: true,
+      isError: false,
+    };
+  }
+
+  componentDidMount() {
+    this.getChartData(this.covidService.getAllDaysGlobal);
+  }
+
+  componentDidUpdate(prevProps) {
+    this.onComponentUpdate(prevProps);
+  }
+
+  onComponentUpdate(prevProps) {
+    const { country } = this.props;
+
+    if (prevProps.country === country) {
+      return;
+    }
+
+    this.setState({
+      isLoading: true,
+      isError: false,
+    });
+
+    if (country) {
+      this.getChartData(this.covidService.getAllDaysByCountry);
+    } else {
+      this.getChartData(this.covidService.getAllDaysGlobal);
+    }
+  }
+
+  onChartDataLoaded = (chartData) => {
+    this.setState({
+      chartData,
+      isLoading: false,
+      isError: false,
+    });
+  }
+
+  onError = () => {
+    this.setState({
+      isError: true,
+      isLoading: false,
+    });
+  }
+
+  getChartData(fn) {
+    const { country } = this.props;
+    fn(country)
+      .then(this.onChartDataLoaded)
+      .catch(this.onError);
+  }
+
+  render() {
+    const {
+      switchersState,
+      onSwitcherChange,
+      startData,
+      country,
+    } = this.props;
+    const { chartData, isLoading, isError } = this.state;
+    const error = isError ? 'No Data' : null;
+    const loading = isLoading ? 'loading...' : null;
+    const app = !(isError || isLoading)
+      ? (
+        <ChartComponentView
+          startData={startData}
+          chartData={chartData}
+          country={country}
+          switchersState={switchersState}
+          onSwitcherChange={onSwitcherChange}
+        />
+      ) : null;
+
+    return (
+      <>
+        { error}
```

Comment:
Add white space before second curly brace here and below

Reply:
Deleted  white spaces after first curly brace as in the whole project

https://github.com/SorokinTim/covid-dashboard-rss/pull/20/commits/a05ba7532e6f46cad9f50b8b7a08051da58bc08a